packages/frontpage/lib/auth.ts

```diff
@@ -562,7 +582,115 @@ export async function fetchAuthenticatedAtproto(
     .set({
       dpopNonce,
     })
-    .where(eq(schema.OauthSession.sessionId, session.user.sessionId));
+    .where(eq(schema.OauthSession.sessionId, session.sessionId));
 
   return response;
 }
+
+type RefreshSessionResult =
+  | { success: true }
+  | {
+      success: false;
+      error:
+        | "NOT_AUTHENTICATED"
+        | "REFRESH_FAILED"
+        | "CONCURRENT_REFRESH_REPLAYED";
+    };
+
+export async function refreshSession(): Promise<RefreshSessionResult> {
```

Comment:
This is a big diff but it's only because this is just been moved from `proxy.ts` to this `refreshSession` function. This is done so sensitive session values (like jwks) don't get returned outside of functions in this module.